compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -1198,6 +1198,11 @@ public void visit(BLangRecordVariable varNode) {
             varNode.setBType(symResolver.resolveTypeNode(varNode.typeNode, env));
         }
 
+        if (checkValidMappingBindingPatternForRecordVariable(varNode.variableList, varNode.typeNode.getBType())) {
```

Comment:
I think it'll be more useful to log the error for the exact field binding pattern, rather than the single error for the mapping binding pattern, so that it's easier to identify which field binding pattern(s) is invalid.

Reply:
This isn't fixed? Basically for each non-rest binding pattern we check if it is not a required field in the RHS expression.

It is not a required field if the mapping value 
- is a map
- is a record with an optional field for the particular key
- is an open record which has a rest field that may include a field for the the mapping value (both `record {| T...; |}` and `record {}`)

This way we don't have to special case the error for maps also.

Reply:
Though you mentioned about the column index of where the error is logged. Earlier it was for the entire expression. Now it points out what are the optional fields